onnxruntime/core/providers/cpu/text/string_normalizer.cc

```diff
@@ -40,8 +32,196 @@ ONNX_CPU_OPERATOR_KERNEL(
+
+// codecvt_utf8 is deprecated, we will want to replace it with our class
+class Utf8ConverterGeneric {
+ public:
+  size_t ComputeRequiredSizeToUtf8(const std::wstring& wstr) const {
+    if (wstr.empty()) {
+      return 0;
+    }
+
+    size_t result = 0;
+    std::mbstate_t state = std::mbstate_t();
+
+    const wchar_t* src = wstr.data();
+    const wchar_t* src_end = src + wstr.length();
+
+    char dummy_dest[128] = {0};
+
+    char* char_next = dummy_dest;
+    const wchar_t* wchar_next = src;
+
+    size_t converted = 0;
+
+    std::codecvt_base::result ret_code = std::codecvt_base::ok;
+
+    // Continue while we exhaust the sequence
+    while (converted < wstr.length()) {
+      ret_code = converter_.out(state,
+                                wchar_next,
+                                src_end,
+                                wchar_next,
+                                std::begin(dummy_dest),
+                                std::end(dummy_dest),
+                                char_next);
+      result += (char_next - dummy_dest);
+      converted = (wchar_next - src);
+
+      if (ret_code != std::codecvt_base::partial &&
+          ret_code != std::codecvt_base::ok) {
+        break;
+      }
+    }
+
+    ORT_ENFORCE(ret_code != std::codecvt_base::noconv, "Conversion is expected");
+
+    if (ret_code != std::codecvt_base::ok) {
+      ORT_THROW("Failed to compute size for UTF-8. Converted only first: ",
+                converted, " codepoints out of: ", wstr.length());
+    }
+
+    return result;
+  }
+
+  // We assume the caller pre-allocated the correct length
+  Status ConvertToUtf8(const std::wstring& wstr, std::string& str) const {
+    if (wstr.empty()) {
+      str.clear();
+      return Status::OK();
+    }
+
+    std::mbstate_t state = std::mbstate_t();
+
+    const wchar_t* src = wstr.data();
+    const wchar_t* src_end = src + wstr.length();
+
+    char* dest = str.data();
+    char* dest_end = dest + str.length();
+
+    char* char_next = dest;
+    const wchar_t* wchar_next = src;
+
+    std::codecvt_base::result ret_code = converter_.out(state,
+                                                        src,
+                                                        src_end,
+                                                        wchar_next,
+                                                        dest,
+                                                        dest_end,
+                                                        char_next);
+
+    if (ret_code != std::codecvt_base::ok) {
+      size_t converted = narrow<size_t>(wchar_next - wstr.data());
+      return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Failed to convert to UTF-8. Converted only first: ",
+                             converted, " codepoints out of: ", wstr.length());
+    }
+
+    str.resize(char_next - dest);
+
+    return Status::OK();
+  }
+
+  Status ComputeRequiredSizeToWideChar(const std::string& str, size_t& wchars) {
```

Comment:
I do not want to check for Status when converting back to Utf8, because we know `wchar_t` is valid. Besides, this is just a size calculation, no way we would ran out of buffer.